src/DataTableCatalog.js

```diff
@@ -0,0 +1,168 @@
+    [cssClasses.ROW_SELECTED]: props.isSelected,
+  }, props.classes);
+  const ariaSelectedValue = props.isSelected ? 'true' : 'false';
+
+  return (
+    <tr
+      data-row-id={props.rowId}
+      className={classes}
+      aria-selected={ariaSelectedValue}
+    >
+      {props.children}
+    </tr>
+  );
+};
+
+const HeaderRowCheckboxCell = (props) => {
+  const { ariaLabel } = props;
+
+  return (
+    <HeaderCell classes={extraCssClasses.HEADER_CELL_CHECKBOX}>
+      <Checkbox classes={cssClasses.HEADER_ROW_CHECKBOX} aria-label={ariaLabel} />
+    </HeaderCell>
+  );
+};
+
+const RowCheckboxCell = (props) => {
+  const { ariaLabelledby } = props;
+
+  return (
+    <Cell classes={extraCssClasses.CELL_CHECKBOX}>
+      <Checkbox classes={cssClasses.ROW_CHECKBOX} isChecked={props.isChecked} ariaLabelledby={ariaLabelledby} />
+    </Cell>
+  );
+};
+
+const DataTable = (props) => {
+  const dataTable = React.createRef();
+
+  useEffect(() => {
+    MDCDataTable.attachTo(dataTable.current);
+  });
+
+  return (
+    <div className={cssClasses.ROOT} ref={dataTable}>
+      <table className='mdc-data-table__table'>
+        <thead>
+          <tr className={cssClasses.HEADER_ROW}>
+            {props.rowSelection ? <HeaderRowCheckboxCell ariaLabel='Toggle all rows' /> : null}
+            {props.data.headers.map((header, index) => <HeaderCell key={index}>{header}</HeaderCell>)}
+          </tr>
+        </thead>
+        <tbody className={cssClasses.CONTENT}>
+          {props.data.rows.map((row, index) => {
+            const isSelected = props.rowSelection ? props.data.selectedRowIndexes.indexOf(index) >= 0 : false;
+            const rowId = `u${idIncrement++}`;
+            return (
+              <Row
+                rowId={rowId}
+                isSelected={isSelected} key={index}>
+                {props.rowSelection ? <RowCheckboxCell ariaLabelledby={rowId} isChecked={isSelected} /> : null}
+                {row.map((content, index) => {
+                  let id = index === 0 ? rowId : null;
+                  return <Cell key={index} id={id} isNumeric={typeof content === 'number'}>{content}</Cell>;
+                })}
+              </Row>
+            );
+          })}
+        </tbody>
+      </table>
+    </div>
+  );
+};
+
+const sampleData = {
+  headers: ['Dessert', 'Calories', 'Fat', 'Carbs', 'Protein (g)'],
+  rows: [
+    ['Frozen yogurt', 159, 6.0, 24, 4.0],
+    ['Ice cream sandwich', 237, 9.0, 37, 4.3],
+    ['Eclair', 262, 16.0, 24, 6.0],
+  ],
+};
+
+const sampleRowSelectionData = {
+  ...sampleData,
+  selectedRowIndexes: [1],
+};
+
+const DataTableHero = () => <DataTable data={sampleData} />;
+const DataTableDemos = () => (
+  <div>
+    <h4 className={'demo-data-table-header'}>Data Table Standard</h4>
+    <DataTable data={sampleData} />
+    <h4 className={'demo-data-table-header'}>Data Table with Row Selection</h4>
+    <DataTable data={sampleRowSelectionData} rowSelection={true} />
```

Comment:
when a prop is a true boolean, you can omit the `={true}`. 

replace `rowSelection={true}` --> `rowSelecton`